Extension/src/LanguageServer/client.ts

```diff
@@ -1017,26 +1061,54 @@ export class DefaultClient implements Client {
             if (index === paths.length - 3) {
                 const result: vscode.Uri[] | undefined = await vscode.window.showOpenDialog();
                 if (result === undefined || result.length === 0) {
-                    if (showSecondPrompt && !compilerDefaults.trustedCompilerFound) {
+                    if (showSecondPrompt && !compilerDefaults.trustedCompilerFound && !statusBarIndicatorEnabled) {
                         this.showPrompt(selectCompiler, true, sender);
                     }
                     action = "browse dismissed";
                     return;
                 }
+                configurationSelected = true;
                 action = "compiler browsed";
                 settings.defaultCompilerPath = result[0].fsPath;
-                ui.showCompilerStatusIcon(false);
             } else {
-                action = "select compiler";
-                settings.defaultCompilerPath = util.isCl(paths[index]) ? "cl.exe" : paths[index];
-                ui.showCompilerStatusIcon(false);
+                configurationSelected = true;
+                if (index < configProvidersIndex && configProviders) {
+                    action = "select config provider";
+                    const provider: CustomConfigurationProvider1 = configProviders[index - 1];
+                    await this.configuration.updateCustomConfigurationProvider(provider.extensionId);
+                    this.onCustomConfigurationProviderRegistered(provider);
+                    telemetry.logLanguageServerEvent("customConfigurationProvider", { "providerId": provider.extensionId });
+                    ui.showConfigureIntelliSenseStatusButton(false, this);
+                    return;
+                } else if (index < compileCommandsIndex) {
+                    action = "select compile commands";
+                    this.configuration.setCompileCommands(this.compileCommandsPaths[index - configProvidersIndex - 1]);
+                    ui.showConfigureIntelliSenseStatusButton(false, this);
+                    return;
+                } else {
+                    action = "select compiler";
+                    settings.defaultCompilerPath = util.isCl(paths[index]) ? "cl.exe" : paths[index];
+                }
             }
 
+            ui.showConfigureIntelliSenseStatusButton(false, this);
             util.addTrustedCompiler(compilerPaths, settings.defaultCompilerPath);
             compilerDefaults = await this.requestCompiler(compilerPaths);
             DefaultClient.updateClientConfigurations();
         } finally {
             telemetry.logLanguageServerEvent('compilerSelection', { action, sender: util.getSenderType(sender)}, { compilerCount: paths.length });
```

Comment:
We need to add a property to differentiate when this is "select a compiler" vs "select IntelliSense Configuration".  Or log a different event.  Ask @AlexandraKemperMS what works better for the experiment.

We need to fix "compilerCount" to be correct.  It's currently reporting the number of items in the quick pick.

Reply:
Sure, I'll look into that.

Reply:
Is what I did okay?

Reply:
yes, it looks good.